### PR TITLE
SqLite Support

### PR DESCRIPTION
**Pull request by MMaximus111**

### CODE REVIEW

File: src/Dommel.Bulk/DatabaseAdapters/SqLiteDatabaseAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        [typeof(bool)] = new GenericTypeMapper<bool>((x, tw) => tw.Write(x ? '1' : '0')),
+        [typeof(byte)] = new GenericTypeMapper<byte>((x, tw) => tw.Write(x)),
+        [typeof(char)] = new GenericTypeMapper<char>((x, tw) => tw.WriteEscapeSqLite(x, true)),
+        [typeof(double)] = new GenericTypeMapper<double>((x, tw) => tw.Write(x.ToString(CultureInfo.InvariantCulture))),
+        [typeof(float)] = new GenericTypeMapper<float>((x, tw) => tw.Write(x.ToString(CultureInfo.InvariantCulture))),
+        [typeof(int)] = new GenericTypeMapper<int>((x, tw) => tw.Write(x)),
+        [typeof(long)] = new GenericTypeMapper<long>((x, tw) => tw.Write(x)),
+        [typeof(decimal)] = new GenericTypeMapper<decimal>((x, tw) => tw.Write(x.ToString(CultureInfo.InvariantCulture))),
+        [typeof(DateTime)] = new GenericTypeMapper<DateTime>((x, tw) => tw.WriteSqLiteDateTime(x, true)),
```
Comment (denisenko93) [inline src/Dommel.Bulk/DatabaseAdapters/SqLiteDatabaseAdapter.cs:20]: SqLite supports date storage in 3 formats. We must document the date format used for insertion

File: src/Dommel.Bulk/DatabaseAdapters/SqLiteDatabaseAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
+﻿using System.Globalization;
+using System.Reflection;
+using Dommel.Bulk.Extensions;
+using Dommel.Bulk.TypeMap;
+
+namespace Dommel.Bulk.DatabaseAdapters;
+
+public class SqLiteDatabaseAdapter : DatabaseAdapterBase
+{
+    private static readonly Dictionary<Type, ITypeMapper> TypeMappers = new Dictionary<Type, ITypeMapper>
```
Comment (denisenko93) [inline src/Dommel.Bulk/DatabaseAdapters/SqLiteDatabaseAdapter.cs:10]: SqLite supports storing an array of bytes in the `BLOB` type. We must implement support for this type

File: src/Dommel.Bulk/DatabaseAdapters/SqLiteDatabaseAdapter.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        var properties = propertiesToUpdate?.ToArray() ?? Array.Empty<PropertyInfo>();
+
+        base.BuildInsertFooter<T>(textWriter, sqlBuilder, flags, properties, constraintName);
+
+        if (properties.Length > 0)
+        {
+            textWriter.WriteLine();
+            textWriter.Write("ON CONFLICT ");
+            textWriter.Write(sqlBuilder.QuoteIdentifier(constraintName));
```
Comment (denisenko93) [inline src/Dommel.Bulk/DatabaseAdapters/SqLiteDatabaseAdapter.cs:66]: You need to check that `constraintName` is not empty

File: src/Dommel.Bulk/Extensions/SqLiteTextWriterExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
+﻿namespace Dommel.Bulk.Extensions;
+
+public static class SqLiteTextWriterExtensions
```
Comment (denisenko93) [inline src/Dommel.Bulk/Extensions/SqLiteTextWriterExtensions.cs]: Extensions must be `internal`

File: tests/Dommel.Bulk.IntegrationTests/Dommel.Bulk.IntegrationTests.csproj [hunk 5]
```diff
@@ -8,6 +8,7 @@
     </PropertyGroup>
 
     <ItemGroup>
+        <PackageReference Include="Microsoft.Data.Sqlite" Version="6.0.5" />
```
Comment (denisenko93) [inline tests/Dommel.Bulk.IntegrationTests/Dommel.Bulk.IntegrationTests.csproj:11]: There are two popular sqlite connectors. System.Data.SQLite.Core and Microsoft.Data.Sqlite.Core. I think it is necessary to write unit tests using both libraries

**Review by denisenko93 — CHANGES_REQUESTED** [review]

File: tests/Dommel.Bulk.IntegrationTests/SqLiteBulkInsertTests.cs [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+using Microsoft.Data.Sqlite;
+using Xunit;
+
+namespace Dommel.Bulk.IntegrationTests;
+
+public class SqLiteBulkInsertTests : BulkInsertTestsBase<SqLiteAllTypesEntity>
+{
+    public SqLiteBulkInsertTests()
+    {
+        SqlMapper.AddTypeHandler(new GuidHandler());
```
Comment (denisenko93) [inline tests/Dommel.Bulk.IntegrationTests/SqLiteBulkInsertTests.cs]: I think this behavior should be documented. You can also add a setting that fixes the problem.

**Review by denisenko93 — CHANGES_REQUESTED** [review]

**Comment by MMaximus111** [comment]
All fixes done.

File: tests/Dommel.Bulk.IntegrationTests/SqLiteBulkInsertTests.cs [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+            await connection.DeleteAllAsync<Person>();
+
+            connection.Dispose();
+        }
+    }
+
+    protected override IEnumerable<IDbConnection> GetConnections()
+    {
+        yield return new SqliteConnection(@"Data Source=C:\Users\Admin\DataGripProjects\Telemart\identifier.sqlite");
```
Comment (denisenko93) [inline tests/Dommel.Bulk.IntegrationTests/SqLiteBulkInsertTests.cs]: Path to identifier.sqlite must be relative

File: tests/Dommel.Bulk.IntegrationTests/BulkInsertTestsBase.cs [hunk 8]
```diff
@@ -11,134 +11,160 @@ namespace Dommel.Bulk.IntegrationTests;
...
 
     [Fact]
     public async Task BulkInsertTestAsync()
     {
-        TAllTypesEntity[] allTypesFromDb;
-
         IReadOnlyCollection<TAllTypesEntity> types = GetAllFakeTypes();
 
-        using (IDbConnection connection = GetOpenConnection())
+        foreach (IDbConnection connection in GetOpenConnections())
```
Comment (denisenko93) [inline tests/Dommel.Bulk.IntegrationTests/BulkInsertTestsBase.cs]: Only one connection should be available in the test case

**Review by denisenko93 — CHANGES_REQUESTED** [review]

**Comment by MMaximus111** [comment]
Done.

**Review by denisenko93 — APPROVED** [review]

**Pull request merged**